sphinx/builders/__init__.py

```diff
@@ -390,6 +383,13 @@ def build(self, docnames, summary=None, method='update'):
                 logger.info(bold(__('no targets are out of date.')))
                 return
 
+        # save the environment
+        from sphinx.application import ENV_PICKLE_FILENAME
+        logger.info(bold(__('pickling environment... ')), nonl=True)
+        with open(path.join(self.doctreedir, ENV_PICKLE_FILENAME), 'wb') as f:
+            pickle.dump(self.env, f, pickle.HIGHEST_PROTOCOL)
+        logger.info(__('done'))
+
```

Comment:
It would be better to move this block before the if-block to keep the processing order. At present, this change invokes the block after consistency-check phase.

Reply:
Done.